gspread/client.py

```diff
@@ -39,15 +37,13 @@ class Client(object):
 
     """
     def __init__(self, auth, session=None):
-        self.auth = auth
-        self.session = session or requests.Session()
+        self.auth = convert_credentials(auth)
+        self.session = session or AuthorizedSession(self.auth)
 
     def login(self):
-        """Authorize client."""
-        if not self.auth.token or (hasattr(self.auth, 'expired') and self.auth.expired):
```

Comment:
@aiguofer I'm fixing the tests now, and I noticed that dropping this condition has increased the number of token refresh requests (there's `login()` in `setUp`). You probably had a good reason for removing the condition, but I can't figure it out. Why is it not needed any longer?

Reply:
The new Google auth library handles refreshes automatically, which renders this `login` function unnecessary. I think I wasn't sure how to handle that when I submitted the PR and I thought it could be used when someone wants to force a refresh for some reason.

We could probably just remove it altogether and add a deprecation notice.